tests/core/test_regions.py

```diff
@@ -521,6 +521,77 @@ def test_pointset_region():
     assert ps.AABB == ((1, 5), (2, 6), (0, 5))
 
 
+def test_voxel_region():
+    encoding = [
+        [[0, 0, 0], [0, 1, 0], [0, 0, 0]],
+        [[0, 1, 1], [0, 1, 0], [1, 1, 0]],
+        [[0, 0, 0], [0, 1, 0], [0, 0, 0]],
+    ]
+
+    vg1 = trimesh.voxel.VoxelGrid(encoding=numpy.asarray(encoding))
+
+    centering_matrix = translation_matrix((vg1.scale - vg1.extents) / 2)
+    vg1.apply_transform(centering_matrix)
+
+    scale = vg1.extents / numpy.array((3, 3, 3))
+    scale_matrix = numpy.eye(4)
+    scale_matrix[:3, :3] /= scale
+    vg1.apply_transform(scale_matrix)
+
+    position_matrix = translation_matrix((4, 5, 6))
+    vg1.apply_transform(position_matrix)
+
+    vr1 = VoxelRegion(vg1)
+
+    assert vr1.containsPoint((4, 5, 6))
+    assert vr1.containsPoint((4, 6, 5))
+    assert vr1.containsPoint((4, 4, 7))
+    assert not vr1.containsPoint((4, 6, 7))
+    assert not vr1.containsPoint((4, 4, 5))
+    assert not vr1.containsPoint((100, 100, 100))
+
+    for _ in range(100):
+        sampled_pt = vr1.uniformPointInner()
+        assert vr1.containsPoint(sampled_pt)
+
+    assert vr1.AABB == ((2.5, 5.5), (3.5, 6.5), (4.5, 7.5))
+
+    vg2 = trimesh.voxel.VoxelGrid(encoding=numpy.asarray(encoding))
+
+    centering_matrix = translation_matrix((vg2.scale - vg2.extents) / 2)
+    vg2.apply_transform(centering_matrix)
+
+    scale = vg2.extents / numpy.array((5, 5, 3))
+    scale_matrix = numpy.eye(4)
+    scale_matrix[:3, :3] /= scale
+    vg2.apply_transform(scale_matrix)
+
+    vr2 = VoxelRegion(vg2)
+
+    assert vr2.size == pytest.approx((7 / 27) * 5 * 5 * 3)
+    assert vr2.dimensionality == 3
+
+
+def test_mesh_voxelization(getAssetPath):
+    plane_region = MeshVolumeRegion.fromFile(getAssetPath("meshes/classic_plane.obj.bz2"))
+    voxel_grid = plane_region.mesh.voxelized(max(plane_region.mesh.extents) / 100).fill()
+    vr = VoxelRegion(voxelGrid=voxel_grid)
```

Comment:
Maybe it would be better to test the `MeshVolumeRegion.voxelized` method here rather than using the raw `trimesh` methods.

Reply:
Good idea!